src/scripts/Update.ts

```diff
@@ -146,24 +146,33 @@ class Update implements Saveable {
                 hoenn: [101, 134],
                 sinnoh: [201, 230],
             };
-            const result = saveData.statistics.routeKills.reduce((acc, nextValue, nextIndex) => {
-                const [region] = Object.entries(regionRoutes).find(([, check]) => (
-                    // Find the region that contains this index
-                    check[0] <= nextIndex && nextIndex <= check[1]
-                )) || ['none'];
-                // Skip over any statistics for the 'none' region that are also 0, since
-                // these are just the gaps in the route numbers
-                if (region === 'none' && nextValue === 0) {
-                    return acc;
-                }
 
-                // Ensure the region has been prepared
-                acc[region] = (acc[region] || {});
-                // Track the route with its number in the statistics
-                acc[region][nextIndex] = nextValue;
-                return acc;
-            }, {});
-            saveData.statistics.routeKills = result;
+            // Re-map route kill statistics
+            if (
+                typeof saveData.statistics.routeKills === 'object' &&
+                Array.isArray(saveData.statistics.routeKills) === false
+            ) {
+                console.log('Skipping 0.6.0 statistics routeKills update because it is already an object');
+            } else {
```

Comment:
I'm not sure about adding this part, seems unnecessary as dev saves shouldn't be main saves, and they can easily be fixed by just manually changing the version number from `0.5.8` -> `0.6.0`.

It's similar to when we add other data that will need to be modified, if the save is already on the latest version, the update functions won't be run, and the save is unlikely to work with the new version.